pkgs/development/compilers/idris2/default.nix

```diff
@@ -1,22 +1,29 @@
-{ lib, stdenv, fetchFromGitHub, makeWrapper
-, clang, chez
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, clang
+, chez
+, gmp
 }:
 
+# NOTICE: An `idris2WithPackages` is available at: https://github.com/claymager/idris2-pkgs
+
 # Uses scheme to bootstrap the build of idris2
 stdenv.mkDerivation rec {
   pname = "idris2";
-  version = "0.3.0";
+  version = "0.4.0";
 
   src = fetchFromGitHub {
     owner = "idris-lang";
     repo = "Idris2";
     rev = "v${version}";
-    sha256 = "0sa2lpb7n6xqfknwld9rzm4bnb6qcd0ja1n63cnc5v8wdzr8q7kh";
+    sha256 = "105jybjf5s0k6003qzfxchzsfcpsxip180bh3mdmi74d464d0h8g";
   };
 
   strictDeps = true;
   nativeBuildInputs = [ makeWrapper clang chez ];
-  buildInputs = [ chez ];
```

Comment:
We remove chez as a runtime dependency, but the wrapper further below does depend on chez:

`--set-default CHEZ "${chez}/bin/scheme"`

Does this work? I know idris binaries are kind of weird wrt the expected existence of chez; do we know the consequences of removing it from buildInput? (I.e., what happens if chez is not separately available or gets garbage collected?)

Reply:
I did test it and buildInputs is not a runtime dependency that would be `propagatedBuildInputs`

Reply:
Changed to `propagatedNativeBuildInputs`